code/headers/test_usart.hpp

```diff
@@ -1,90 +1,97 @@
 #pragma once
+/**
+ * @file test_usart.hpp
+ * @author Patrick Dekker
+ * @brief Catch2 Test implementation for the usart
+ * @version 0.1
+ * @date 2019-05-24
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
 
-#include <vector>
 #include <string>
-#include <queue.hpp>
+#include <vector>
+
 #include <usart_connection.hpp>
 
 namespace r2d2::usart {
-
-    template <size_t BufferLength = 250>
+    /**
+     * @brief Usart implementation to test the usart with Catch2
+     *
+     */
     class test_usart_c : public usart_connection_c {
     private:
-        queue_c<uint8_t, BufferLength> receive_buffer;
+        std::vector<uint8_t> receive_buffer;
```

Comment:
I do not know this for sure, but does the vector play nice with the due? since as far as I remember vectors need to be dodged on the due.

Reply:
The test usart is not intended to run on the arduino due. It is only intended to use with Catch2. This change to a std::vector is so the template can be removed from the test_usart

Reply:
fair enough did not see that.